include/clap/factory/draft/preset-discovery.h

```diff
@@ -196,7 +199,11 @@ typedef struct clap_preset_discovery_location {
    uint32_t    flags;    // see enum clap_preset_discovery_flags
    const char *name;     // name of this location
    uint32_t    kind;     // See clap_preset_discovery_location_kind
-   const char *location; // Actual location in which to crawl presets
+
+   // Actual location in which to crawl presets.
+   // For FILE kind, the location can be either a path to a directory or a file.
+   // For PLUGIN kind, the location is arbitrary.
```

Comment:
Does arbitrary mean that the parameter should be ignored, or that the host should repeat the value from the `location` field verbatim?

Reply:
The current idea is that the provider can use whatever it wants for the location, and then the indexer/host just give that string back.

I think it can lead to duplicates, if the plugin is installed twice with and two providers uses different location for the same plugin target.

While if the plugin is the preset container, then the location should be irrelevant. I'm very tempted to force it to be null.

Reply:
Let's force it to null for now, and if anyone needs it then we re-visit it.

Reply:
> The current idea is that the provider can use whatever it wants for the location, and then the indexer/host just give that string back.

Yeah that makes sense to me, probably worth explicitly mentioning that the arbitrary string should still be used by the host and not treated like arbitrary data that can be ignored. I don't think allowing locations can lead to more duplicates than not allowing locations though. The location is still relative to the plugin's shared library.

Reply:
Yeah but if the benefit is 100% hypothetical, then let's get rid of it.

Reply:
I force the null location.